src/main/java/frc/robot/pose/RobotPose.java

```diff
@@ -0,0 +1,51 @@
+package frc.robot.pose;
+
+import edu.wpi.first.wpilibj.geometry.Pose2d;
+import edu.wpi.first.wpilibj.geometry.Rotation2d;
+import frc.robot.utils.VisionDataProcessor;
+
+public class RobotPose {
+
+    private RobotPosition robotPosition;
+    private VisionData visionData;
+    private WheelColorValue wheelColor;
+    private TargetLocation targetLocation;
+
+    public RobotPose(RobotPosition robotPosition, VisionData vData) {
+        this.robotPosition = robotPosition;
+        visionData = vData;
+        wheelColor = null;
+        targetLocation = new VisionDataProcessor().compute(vData);
+    }
+
+    public RobotPose(RobotPosition robotPos) {
+        robotPosition = robotPos;
+        visionData = null;
+        wheelColor = null;
+        targetLocation = new VisionDataProcessor().compute(visionData);
+    }
+
+    public RobotPosition getRobotPosition() {
+        return robotPosition;
+    }
+
+    public Pose2d getWPIRobotPose() {
```

Comment:
duplicated  in RobotPosition.getWpiRobotPose().  this one probbably needs to go. Note the code smell here-- every seingle field is coming from robotPosition-- so clearly robotPosition is the right place for it